src/generator/GAN-generator.py

```diff
@@ -0,0 +1,249 @@
+                    momentum: variable for batch normalization
+                
+                Returns:
+                    Keras layer
+            """
+            for _ in range(number):
+                layer = residual_block(layer, filters, momentum)
+            return layer
+        
+        def subpixel_block(layer, filters, momentum):
+            """
+                sub-pixel block consisting of
+                    Conv2D -> pixel shuffler x 2 -> Batch Normalization -> Relu
+                    
+                the code of subpixel layer is based on https://github.com/Tetrachrome/subpixel
+                    
+                Args:
+                    layer:   Keras Layer
+                    filters: output size as an integer
+                    momentum: variable for batch normalization
+                
+                Returns:
+                    Keras layer
+            """  
+            
+            layer = Conv2D(filters=filters, kernel_size=(3, 3), strides=(1, 1), padding="same")(layer)
+            layer = Subpixel(filters, (3,3), 2)(layer)
+
+            layer = BatchNormalization(momentum= momentum)(layer)
+            layer = Activation('relu')(layer)
+            return layer
+        
+        def subpixel_layer(layer, number, filters, momentum):
+            """
+                Facade for subpixel block.
+                
+                Creates subpixel layer with specified number of subpixel blocks
+                
+                Args:
+                    layer:   Keras layer
+                    number:  number of subpixel blocks in layer
+                    filters: output size as an integer
+                    momentum: variable for batch normalization
+                
+                Returns:
+                    Keras layer
+            """
+            for _ in range(number):
+                layer = subpixel_block(layer, filters, momentum)
+            return layer
+        inputs = Input(shape=input_shape)
+        filters = INITIAL_FILTERS
+        layer = Dense(DEPTH*DIM*DIM)(inputs)
+
+        layer = BatchNormalization(momentum = MOMENTUM)(layer)
+        layer = Activation('relu')(layer)
+        layer = Reshape((DIM,DIM,DEPTH))(layer)
+        old = layer
+        print("starting layer built")
+        # 16 residual layers
+        layer = residual_layer(layer, NUM_RESIDUAL, filters, MOMENTUM)
+    
+        
+        layer = BatchNormalization(momentum = MOMENTUM)(layer)
+        layer = Activation('relu')(layer)
+        layer = Add()([layer, old])
+        
+        print("residual layer built")
+
+        filters *= 4
+        # 3 sub-pixel layers
+        layer = subpixel_layer(layer, NUM_SUBPIXEL, filters, MOMENTUM)
+
+        print("sub-pixel layer built")
+        
+        layer = Conv2D(filters=FINAL_FILTERS, kernel_size=(9, 9), strides=(1, 1), padding="same")(layer)
+        layer = Activation('tanh')(layer)
+        
+        print("final layer built")
```

Comment:
Remove print statements. You can leave it in the Ipython notebook but clean up the raw code.